### PR TITLE
Add support for creating internal networks

### LINKED ISSUE #3190: Ability to create internal networks missing
I cannot find a way to specify in the docker-compose file (even v2) that a network must be internal (as if it was created with `docker network create --internal ...`).

### PR DESCRIPTION
**Pull request by jgiannuzzi**
Fixes #3190

### CODE REVIEW

**Comment by shin-** [comment]
Thanks!

Oh man, this is going to be great...
```yaml
networks:
  foo:
    internal: true
    external: true
```

All jokes aside, code LGTM!

**Comment by jgiannuzzi** [comment]
Rebased onto master

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
 PyYAML==3.11
 cached-property==1.2.0
-docker-py==1.8.1
 dockerpty==0.4.1
 docopt==0.6.1
 enum34==1.0.4
+git+https://github.com/docker/docker-py.git@26f2b696ab61a6dd41a19c9730f85a6b572089b7#egg=docker-py
 jsonschema==2.5.1
```
Comment (denismakogon) [inline requirements.txt]: Hi there, i'm not really sure that we should do something like this, can't we wait until docker-py will support internal networks (until new version will see the light)?

**Comment by denismakogon** [comment]
LGTM +1

File: requirements.txt (hunk 1 above)
Comment (jgiannuzzi) [inline requirements.txt]: Well, that's what @shin- suggested here: https://github.com/docker/docker-py/pull/1068#issuecomment-220461252

Comment (shin-) [inline requirements.txt]: Yes, we do this all the time, then switch back to a release version when moving closer to release date.

**Comment by jgiannuzzi** [comment]
Oops, I did something stupid and pressed the "Update branch" button in GitHub, and then something even more stupid, I deleted the branch.

Is there any way to open this PR again?

**Comment by aanand** [comment]
LGTM

Unfortunately we're past code freeze, so I've moved this into the 1.9.0 milestone.

**Comment by jgiannuzzi** [comment]
Thanks, @aanand!

**Comment by peterthomassen** [comment]
The new paragraph in the documentation (https://github.com/docker/compose/pull/3488/files#diff-b94299404e9b589177d56080e454dac3R862) sounds like `internal` only is usable in combination with `overlay` networks. Is that true?

It looks like the paragraph was half taken from https://docs.docker.com/engine/reference/commandline/network_create/#network-internal-mode where the other half may have been relevant as well.

**Pull request merged**